### PR TITLE
Bug 2091746: Check if spec is available in MCP details page

### PR DESCRIPTION
**Pull request by jhadvig**
/assign @TheRealJon 
/cherry-pick release-4.10

### CODE REVIEW

**Comment by rhamilto** [comment]
https://github.com/openshift/console/blob/0aead8fb830cdf41451663fe7aae1e74c6eec72f/frontend/public/components/utils/workload-pause.tsx#L13 also needs to be updated.

**Comment by rhamilto** [comment]
Also see https://github.com/openshift/console/pull/11502#issuecomment-1142145420

**Comment by rhamilto** [comment]
> https://github.com/openshift/console/blob/0aead8fb830cdf41451663fe7aae1e74c6eec72f/frontend/public/components/utils/workload-pause.tsx#L13
> 
> also needs to be updated.

I think we want the following:
```
export const togglePaused = (model: K8sKind, obj: K8sResourceKind) => {
  if (!obj.spec) {
    const newObj = _.cloneDeep(obj);
    newObj.spec = { paused: true };

    return k8sUpdate(model, newObj);
  }

  const patch = [
    {
      path: '/spec/paused',
      op: 'add',
      value: !obj.spec.paused,
    },
  ];

  return k8sPatch(model, obj, patch);
};
```

File: frontend/public/module/k8s/types.ts [hunk 1]
```diff
@@ -753,19 +753,22 @@ export type MachineConfigPoolStatus = {
...
-  conditions: MachineConfigPoolCondition[];
+  conditions?: MachineConfigPoolCondition[];
 };
 
 export type MachineConfigPoolSpec = {
   machineConfigSelector?: Selector;
   maxUnavailable?: number | string;
   nodeSelector?: Selector;
-  paused: boolean;
+  paused?: boolean;
```
Comment (andrewballantyne) [inline frontend/public/module/k8s/types.ts:763]: Is this optional when `spec` is there -- or is this a result of "if spec is optional, all children are optional"?

File: frontend/public/module/k8s/types.ts [hunk 2]
```diff
@@ -753,19 +753,22 @@ export type MachineConfigPoolStatus = {
...
 };
 
 export type MachineConfigPoolKind = {
+  // spec is required per
+  // https://github.com/openshift/machine-config-operator/blob/4fbbead6d1a6e83d9736e75cdc359e10d5adfb84/pkg/apis/machineconfiguration.openshift.io/v1/types.go#L228-L229
+  // but the API doesn't enforce it
   spec: MachineConfigPoolSpec;
```
Comment (rhamilto) [inline frontend/public/module/k8s/types.ts:770]:
```suggestion
  spec?: MachineConfigPoolSpec;
```

**Review by andrewballantyne — CHANGES_REQUESTED** [review]

File: frontend/public/module/k8s/types.ts (hunk 1 above)
Comment (rhamilto) [reply frontend/public/module/k8s/types.ts:763]: My change is to cover the latter.  https://github.com/openshift/machine-config-operator/blob/4fbbead6d1a6e83d9736e75cdc359e10d5adfb84/pkg/apis/machineconfiguration.openshift.io/v1/types.go#L235-L262 doesn't specify any as required or optional.

File: frontend/public/components/utils/workload-pause.tsx [hunk 3]
```diff
@@ -1,11 +1,24 @@
 import * as React from 'react';
+import * as _ from 'lodash-es';
```
Comment (rhamilto) [inline frontend/public/components/utils/workload-pause.tsx:2]: This is no longer needed.

**Comment by jhadvig** [comment]
There was an error in json format. Adding label manually.

File: frontend/public/module/k8s/types.ts [hunk 4]
```diff
@@ -753,19 +753,24 @@ export type MachineConfigPoolStatus = {
...
   maxUnavailable?: number | string;
   nodeSelector?: Selector;
-  paused: boolean;
+  paused?: boolean;
 };
 
 export type MachineConfigPoolKind = {
-  spec: MachineConfigPoolSpec;
-  status: MachineConfigPoolStatus;
+  /*
```
Comment (andrewballantyne) [inline frontend/public/module/k8s/types.ts:767]: FWIW -- JSDoc comments start with two `*` (`/** JSDoc */`, `/* Block comment */`)

**Pull request merged**